src/utils/time.ts

```diff
@@ -61,33 +60,76 @@ export function normalizeTime(date: Date, components: TimeComponents) {
         newDate.setSeconds(components.seconds > 0 ? Math.floor(newDate.getSeconds() / components.seconds) * components.seconds : components.seconds);
     }
     if (components.ms >= 0) {
-        newDate.setMilliseconds(components.ms > 0 ? Math.floor(newDate.getSeconds() / components.ms) * components.ms : components.ms);
+        newDate.setMilliseconds(components.ms > 0 ? Math.floor(newDate.getMilliseconds() / components.ms) * components.ms : components.ms);
     }
     return newDate
 }
 
-enum Seconds {
-    MINUTE=60,
+export enum Milliseconds {
+    SECOND=1000,
+    MINUTE=SECOND*60,
     HOUR=MINUTE*60,
-    // DAY=HOUR*24,
-    // MONTH=DAY*30,
-    // YEAR=MONTH*12,
+    DAY=HOUR*24,
+    WEEK=DAY*7,
+    MONTH=DAY*30,
+    QUARTER=MONTH*3,
+    YEAR=DAY*365,
 }
 
-export function toMilliseconds(components: TimeComponents) {
+export function toMilliseconds(components: DateTimeComponents) {
```

Comment:
It's likely not practically relevant, but should we not at least give a hint via JSDoc on the interfaces, that we cannot handle arithmetics on, say
```typescript
<DateTimeComponents> {
  ms: Number.MAX_VALUE,
  seconds: Number.MAX_VALUE,
}
```
or any other case where the sum of fields overflows?

Reply:
I am not entirely current with TypeScript features, but I believe we cannot feasibly restrict the domain of our interfaces to values we can process as-expected.

Given the presumably exotic nature of this error class, I'd not, on the other hand, want to introduce actual runtime checks to watch out for it.

So documentation of the _actually intended_ domain seems to be the sensible (middle) ground.

Reply:
I don't want to taint my code for except unusual edge cases like Number overflows.
```
Number.MAX_VALUE*2 // Infinity
```

Reply:
Yes, me neither. Would be more code for the edge case than for the normal behaviour.

That's why I recommend putting an information as JSDoc onto the three interfaces: "Values should not sum to Number.MAX_VALUE or Number.MIN_VALUE".

That way it's just an RTFM and no longer our concern.

Reply:
`Number.MAX_VALUE` would be `1.7976931348623157e+305` seconds. I dont know if the universe will still exist then.

Reply:
No, true, it will never show up as result of a meaningful calculation. It might be a consecutive fault, though. But very well.